dataprofiler/data_readers/csv_data.py

```diff
@@ -186,7 +198,7 @@ def _guess_delimiter_and_quotechar(
         vocab = Counter(data_as_str)
         if "\n" in vocab:
             vocab.pop("\n")
-        for char in omitted + [quotechar]:
+        for char in omitted + ([quotechar] if quotechar is not None else []):
```

Comment:
I think this line gets a little confusing if it's all in one. If this if is necessary, can we do it before the `for`?
```python
omitted_list: list[str] = ommitted
if quotechar is not None:
    omitted_list: list[str] = omitted + [quotechar]
```

Reply:
then we can use that in the for

Reply:
@Sanketh7 in case you missed

Reply:
Done